.github/workflows/container.yml

```diff
@@ -0,0 +1,43 @@
+name: Create and publish a container image
+
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - "*"
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build-and-push-image:
+    name: Build and push container image
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Log in to the Container registry
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Extract metadata (tags, labels)
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+      - name: Build and push container image
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          push: ${{ github.event_name != 'pull_request' }}
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
```

Comment:
@rhtenhove are these `env.` and `secrets.` elements going to be problematic for those who fork this repo?

Are they going to be problematic for me? 😄

Reply:
The job only runs on `master` and on tags, i.e. people forking it correctly should not ever see this job run on their own fork.
For those that do push to their own `master`, will have the job run with their own credentials, pushing the image to their own registry. Also not a problem, because when they delete their fork after sending in a PR, the images will be gone as well.

The secret used here is an automatic one, made available to any GHA workflow with proper access restrictions, so only pushes done by you (or anyone else with full access to your repo) to `master` will actually build an image and push it to the ghcr.io/StevenBlack/hosts registry.

If anything goes wrong, the job will fail and no harm done, we can fix it later. For example, I'm unsure if it will complain about using capital letters in the repo name.

Reply:
Thank you for this clarification @rhtenhove.

I'm wary of introducing dependencies, especially environment variables.

Also, small point: making Pull Requests is one of many reasons people fork a Github repository, and the presumption that folks delete forks once PRs are done doesn't hold, at least in my experience.

Reply:
The `env` key is introduced in the file itself (at the top) and is about the environment of only the job in that file. The secret GITHUB_TOKEN is also a convenient token, as it ensures it'll behave the same for everyone, without having to set anything up.

Finally, forking and then deleting or not doesn't matter, so my presumption is actually irrelevant. It is meant to clarify that it will not have _any_ consequences for people forking, and nobody will ever need to do anything with the content or results of that job if they don't wish to.

That's so great about CI and the provided default variables :-)